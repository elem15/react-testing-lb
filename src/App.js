import logo from './logo.svg';
import './App.css';
import { useState } from 'react';


const Search = ({ value, onChange, children }) => {
  return (
    <form>
      <label>{children}</label>
      <input placeholder='search' type='search' name='search' value={value} onChange={onChange} />
    </form>
  )
}
function App() {
  const [search, setSearch] = useState('');
  const onChange = ({ target }) => {
    setSearch(target.value);
  }
  return (
    <div className="App">
      <Search value={search} onChange={onChange}>
        Search
        <br/>
      </Search>
      <div>
        Search for: {search ? search : '...'}
      </div>
    </div>
  );
}

export default App;
