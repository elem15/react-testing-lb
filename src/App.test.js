import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

test('renders learn react link', () => {
  const { asFragment } = render(<App />);
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();

  expect(asFragment(<App />)).toMatchSnapshot();
});
