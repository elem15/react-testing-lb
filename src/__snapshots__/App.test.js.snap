// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <form>
      <label>
        Search
        <br />
      </label>
      <input
        name="search"
        placeholder="search"
        type="search"
        value=""
      />
    </form>
    <div>
      Search for: ...
    </div>
  </div>
</DocumentFragment>
`;
